src/guiguts/maintext.py

```diff
@@ -2902,8 +2928,14 @@ def highlight_selection(
                     tag_name, match.rowcol.index(), match.rowcol.index() + "+1c"
                 )
 
+    def remove_search_highlights(self) -> None:
+        """Remove highlights for search"""
+        if maintext().winfo_exists():
```

Comment:
Sorry @tangledhelix - I misled you. I think this fix would actually be better if it said
```python
        if self.winfo_exists():
```
If you agree, please change it. If you think it doesn't matter, don't worry. I'm going to approve anyway now.